src/common/odometer.cpp

```diff
@@ -1,15 +1,15 @@
-// odometer.cpp
+// odometer_s.cpp
 
 #include <cmath>
 
 #include "odometer.hpp"
 #include "cmath_ext.h"
 #include "eeprom.h"
 
-odometer_c odometer;
+Odometer_s &odometer_s(Odometer_s::instance());
```

Comment:
Making a top level reference of a singleton's instance is a bad practice. Makes the linker call/initializace the instance before anything else. Please use the singleton properly at the spots where needed.